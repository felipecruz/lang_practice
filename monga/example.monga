12902       /* TK_NUMBER */
10          /* TK_NUMBER */

a           /* TK_ID */

0xFF        /* TK_HEXA */
0x00        /* TK_HEXA */
0xGG        /* TK_NUMBER ; TK_ID */

"string simples"
"string com \n barra n"
"string com \t barra t"
