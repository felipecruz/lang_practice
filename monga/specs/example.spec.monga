void f () {
    a = 12902;       /* TK_NUMBER */

    a = 1;           /* TK_ID */

    st = "string simples"; /* TK_STRING */
}
