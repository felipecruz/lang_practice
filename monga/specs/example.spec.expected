
 Func Decl - Name: f Return Type: Void 
          - Paramenters 
   Var Decl Int Name a 
   Var Decl Char[] Name st 
   Assignment Var: a <- EXP: Const int: 12902
   Assignment Var: a <- EXP: Const int: 1
   Assignment Var: st <- EXP: Const string: string simples
